### PR TITLE
[20091] Add macOS to gihub actions testing

### PR DESCRIPTION
**Pull request by EduPonz**
This PR adds macOS 13 to the list of platforms tested in Github CI. Depends on:

* eProsima/eProsima-CI#44

@mergifyio backport 1.1.x 1.0.x

### CODE REVIEW

File: .github/workflows/fastcdr-test.yml
Comment (JLBuenoLopez-eProsima) [inline .github/workflows/fastcdr-test.yml:63]: Just to ensure that before merging a release have been done and this tag updated.

File: .github/workflows/fastcdr-test.yml
Comment (JLBuenoLopez-eProsima) [inline .github/workflows/fastcdr-test.yml:59]: Fast DDS MacOS official Tier 1 platform is MacOS 10.14 (Mojave). ROS 2 demoted MacOS to Tier 3 with Galactic and since then, they do not specify the platform supported (it is community supported only). I suppose we should update the platform support in Fast DDS.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
LGTM though I will approve after the tags have been updated with the proper eProsima CI release.

File: .github/workflows/fastcdr-test.yml
Comment (EduPonz) [reply .github/workflows/fastcdr-test.yml:59]: I think we need to discuss this internally and update here on a follow up if needed

**Review by JesusPoderoso — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/381/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/283/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/361/)

**Pull request merged**
